### DISCUSSION TITLE
Looping disconnected from slides

### DISCUSSION THREAD

**Discussion by Adcombrink** [jeertmans/manim-slides#301] [Ideas]
Loving manim-slides so far, great work. However, I've been trying to do something and am starting to think it's not possible, maybe a new feature. I want to click to the next slide, have an animation play, and then loop another animation, all with only clicking the first time. Right now, if i write 

```python
self.next_slide()

self.play(... initial animations ...)
self.start_loop()
self.play(... looping animations ...)
self.end_loop()
``` 
then it instantly jumps to the looping part, as if the initial animations have already run.

**Comment by jeertmans** [comment]
Hello, indeed this is not possible.
I could add a « auto-skip-slide » flag and watch of the end of media to skip a slide. But don’t know if this is best to do that in the general config file (via next_slide), or via the command line options when presenting.

**Reply by Adcombrink** [reply]
Hi! If I understand correctly, then I think it would be cumbersome to have that via the command line. Personally, I would prefer to have it as an argument to ```self.next_slide()``` , for example ```self.next_slide(auto_next=True)```or something similar. 

Also, in the [documentation](https://eertmans.be/manim-slides/reference/api.html), it says that ```next_slide``` takes a keyword argument ```bool loop```, however that does not work for me and therefore I use ```start_loop()``` and ```end_loop()```. Is the keyword ```loop``` new, or is it an old and unsupported feature?

**Reply by jeertmans** [reply]
> Hi! If I understand correctly, then I think it would be cumbersome to have that via the command line. Personally, I would prefer to have it as an argument to self.next_slide() , for example self.next_slide(auto_next=True)or something similar.

I agree with that, I'll see how to eventually implement it :-)

> Also, in the [documentation](https://eertmans.be/manim-slides/reference/api.html), it says that next_slide takes a keyword argument bool loop, however that does not work for me and therefore I use start_loop() and end_loop(). Is the keyword loop new, or is it an old and unsupported feature?

Yes, the documentation is for v5 (which is in release mode currently), and not for v4. Hopefully, v5 will be soon available, so you will be noticed.

**Reply by Adcombrink** [reply]
Looking forward to v5, and thank you for the work you are doing!

**Reply by jeertmans** [reply]
Thanks :D

**Reply by jeertmans** [reply]
For triage purposes, this has been implemented as part of v5: https://github.com/jeertmans/manim-slides/releases/tag/v5.0.0.

**Reply by Adcombrink** [reply]
Nice! :D
